src/mpi/comm/comm_split_type.c

```diff
@@ -580,14 +759,8 @@ int MPIR_Comm_split_type_neighborhood(MPIR_Comm * comm_ptr, int split_type, int
         MPIR_Comm_get_ptr(dummycomm, dummycomm_ptr);
         *newcomm_ptr = dummycomm_ptr;
 #endif
-    } else {
-        /* In the mean time, the user passed in
-         * COMM_TYPE_NEIGHBORHOOD but did not give us an info we
-         * know how to work with.  Throw up our hands and treat it
-         * like UNDEFINED.  This will result in MPI_COMM_NULL
-         * being returned to the user. */
-        *newcomm_ptr = NULL;
-    }
+    } else
```

Comment:
A style nitpick - I prefer to leave the braces here. If we use braces for one part of a conditional, we should use them for all parts.

Reply:
Fixed this